src/Adapter/DoctrineItems.php

```diff
@@ -160,16 +167,17 @@ public function list(int $limit = 10, ?string $cursor = null) : ItemListPage
             ->orderBy('item.sequence')
             ->setMaxResults($limit + 1);
 
-        if ($cursor) {
+        if (is_string($cursor)) {
             $query
                 ->where($query->expr()->gte('item.sequence', ':cursor'))
                 ->setParameter('cursor', (int) $cursor);
         }
 
+        /** @var Statement $results */
         $results = $query->execute();
         $ids = $results->fetchAll(FetchMode::ASSOCIATIVE);
 
-        if (empty($ids)) {
+        if (0 === count($ids)) {
```

Comment:
since the return type is `array`, consider `[] == $ids` which doesn't involve `count()`